lib/cli-config.js

```diff
@@ -70,12 +72,16 @@ exports.getContent = function(config, directory) {
 
         if (ext === '.js' || ext === '.json') {
             content = require(configPath);
+        } else if (ext === '.yaml') {
+            content = yaml.safeLoad(fs.readFileSync(configPath, 'utf8'));
         } else {
-            content = JSON.parse(
-                stripJSONComments(
-                    fs.readFileSync(configPath, 'utf8')
-                )
-            );
+            data = fs.readFileSync(configPath, 'utf8');
+
+            try {
+                content = JSON.parse(stripJSONComments(data));
+            } catch (e) {
+                content = yaml.safeLoad(data);
```

Comment:
Oh, I'm a big fan of jsonlint. All for it. But that seems to be outside of the scope of this PR. This PR with what I'm proposing does not make the error more cryptic. Fair enough, "parse error" is cryptic, but that's always been the case here. Is this something we can iterate on?